tools/makecsr/README.md

```diff
@@ -1,8 +1,52 @@
-makecsr
-=======
+# makecsr
 
-This is a small helper used to generate a CSR from a private key hosted in HSM.
-It's used particularly to issue the CS and AMO intermediates used by autograph.
+This is a small helper used to generate a PEM-encoded CSR from a private key
+hosted in our HSMs. It's used particularly to issue the CS and AMO intermediates
+used by autograph. See our private hsm repo for how we've invoked it.
```

Comment:
It doesn't necessarily need to be noted here, but I assume we use this for any CSRs we may need from non-exported HSM keys? (eg: authenticode)

Reply:
The only documented use was for creating intermediates from the addons/content signature root. 

I've not yet seen what templating Digicert and authenticode needs in its CSR, however. The mozilla-services/hsm scripts do a lot of overriding attributes in the CSR that I imagine Digicert wouldn't.